lib/Sema/CSSimplify.cpp

```diff
@@ -9864,13 +9864,23 @@ bool ConstraintSystem::resolveClosure(TypeVariableType *typeVar,
         auto flags = param.getParameterFlags();
 
         // Note when a parameter is inferred to be isolated.
-        if (contextualParam->isIsolated() && !flags.isIsolated() && paramDecl)
+        if (contextualParam->isIsolated() && !flags.isIsolated() && paramDecl) {
           isolatedParams.insert(paramDecl);
 
+          // an 'isolated' param is also implicitly '_local',
+          // since in order to run "on" an actor, it must be a real local instance.
+          distributedKnownLocalParams.insert(paramDecl);
```

Comment:
Hmm. This feels like a semantic property that should be checked elsewhere, e.g., a "known to be local" predicate that checks both `isolated` and `_local`, rather than inserting into both sets at this point. (Something we likely won't do consistently everywhere).

Reply:
Not sure I understood how to take action on this bit -- agreed, but where would I centralize this? 👍